### PR TITLE
Added Azure.MariaDB.GeoRedundantBackup

### LINKED ISSUE #1848: Azure Database for MariaDB servers should should store backups in a geo-redundant storage
# Rule request

## Suggested rule change

Lets add support for the relational database service based on the open-source MariaDB Server engine since we already have MySQL and PostgreSQL. This works similar. 

Reliability pillar suits this rule.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.DBforMariaDB/servers]**

## Additional context

[Backup and restore in Azure Database for MariaDB](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)

### PR DESCRIPTION
**Pull request by bengeset96**
## PR Summary

Fixes #1848

Added Added Azure.MariaDB.GeoRedundantBackup rule.

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/en/rules/Azure.MariaDB.GeoRedundantBackup.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+## NOTES
+
+This rule is only applicable for Azure Database for Maria DB Servers with `General Purpose` and `Memory Optimized` tiers. The `Basic` tier does not support geo-redundant backup storage.
+
+## LINKS
+
+- [Target and non-functional requirements](https://learn.microsoft.com/azure/architecture/framework/resiliency/design-requirements)
+- [Backup and restore in Azure Database for MariaDB](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.MariaDB.GeoRedundantBackup.md:104]:
```suggestion
- [Backup and restore in Azure Database for MariaDB](https://learn.microsoft.com/azure/mariadb/concepts-backup)
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 Looks great. Just one correction.

**Comment by bengeset96** [comment]
@BernieWhite 

Adjusted 😃

**Review by BernieWhite — APPROVED** [review]
Thanks @bengeset96. All good to merge.

**Pull request merged**